geo/src/main/java/org/apache/carbondata/geo/GeoHashImpl.java

```diff
@@ -0,0 +1,400 @@
+    }
+    String GRID_SIZE = commonKey + "gridsize";
+    String gridSize = properties.get(GRID_SIZE);
+    if (StringUtils.isEmpty(gridSize)) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, GRID_SIZE));
+    }
+    String CONVERSION_RATIO = commonKey + "conversionratio";
+    String conversionRatio = properties.get(CONVERSION_RATIO);
+    if (StringUtils.isEmpty(conversionRatio)) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, CONVERSION_RATIO));
+    }
+
+    // Fill the values to the instance fields
+    this.oriLatitude = Double.valueOf(originLatitude);
+    this.userDefineMaxLongitude = Double.valueOf(maxLongitude);
+    this.userDefineMaxLatitude = Double.valueOf(maxLatitude);
+    this.userDefineMinLongitude = Double.valueOf(minLongitude);
+    this.userDefineMinLatitude = Double.valueOf(minLatitude);
+    this.gridSize = Integer.parseInt(gridSize);
+    this.conversionRatio = Integer.parseInt(conversionRatio);
+    this.lon0ByRation = userDefineMinLongitude * this.conversionRatio;
+    this.lat0ByRation = userDefineMinLatitude * this.conversionRatio;
+    calculateData();
+  }
+
+  /**
+   * Generates the GeoHash ID column value from the given source columns.
+   * @param sources Longitude and Latitude
+   * @return Returns the generated hash id
+   * @throws Exception
+   */
+  @Override
+  public String generate(List<?> sources) throws Exception {
+    if (sources.size() != 2) {
+      throw new RuntimeException("Source columns list must be of size 2.");
+    }
+    if (sources.get(0) == null || sources.get(1) == null) {
+      // Bad record. Just return null
+      return null;
+    }
+    if (!(sources.get(0) instanceof Long) || !(sources.get(1) instanceof Long)) {
+      throw new RuntimeException("Source columns must be of Long type.");
+    }
+    Long longitude = (Long) sources.get(0);
+    Long latitude  = (Long) sources.get(1);
+    // generate the hash code
+    int[] gridPoint = calculateID(longitude, latitude);
+    Long hashId = createHashID(gridPoint[0], gridPoint[1]);
+    return String.valueOf(hashId);
+  }
+
+  /**
+   * Query processor for GeoHash.
+   * example: POLYGON (35 10, 45 45, 15 40, 10 20, 35 10)
+   * so there will be a sample check
+   * @param polygon a group of pints, close out to form an area
+   * @return Returns list of ranges of GeoHash IDs
+   * @throws Exception
+   */
+  @Override
+  public List<Long[]> query(String polygon) throws Exception {
+    if (!validate(polygon)) {
+      return null;
+    } else {
+      String[] pointList = polygon.trim().split(",");
+      List<double[]> queryList = new ArrayList<>();
+      for (String str: pointList) {
+        String[] points = str.split(" ");
```

Comment:
Need to trim() before split here. Otherwise fails if there is a space after comma between pair of long lat.
This is handled in `validate()` already. 
Also please split here with regex like this - `str.trim().split("\\s+");` Otherwise if we have more space or tab widths between longitude and latitude, it fail.

Reply:
done